### PR TITLE
MGMT-16791: disable VipAutoAllocation for 4.15 and above

### PR DESCRIPTION
**Pull request by rccrdpccl**
Disable vip auto allocation for 4.15 and later
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by gamli75 — APPROVED** [review]

File: internal/featuresupport/features_networking.go [hunk 1]
```diff
@@ -458,3 +457,11 @@ func getNetworkType(cluster *common.Cluster, updateParams *models.V2ClusterUpdat
...
 	return networkType
 }
+
+// Returns true if openshiftVersion is less than targetVersion or undefined
+func openshiftVersionLessThan(targetVersion, openshiftVersion string) bool {
+	if isAvailable, err := common.BaseVersionLessThan(targetVersion, openshiftVersion); openshiftVersion == "" || err == nil && isAvailable {
+		return true
+	}
+	return false
+}
```
Comment (adriengentil) [inline internal/featuresupport/features_networking.go:467]:
```suggestion
func openshiftVersionLessThan(targetVersion, openshiftVersion string) bool {
	isAvailable, err := common.BaseVersionLessThan(targetVersion, openshiftVersion)
        return openshiftVersion == "" || (err == nil && isAvailable)
}
```
nit

**Review by adriengentil — APPROVED** [review]
Looks good, just a nit

File: internal/featuresupport/features_networking.go [hunk 2]
```diff
@@ -458,3 +457,10 @@ func getNetworkType(cluster *common.Cluster, updateParams *models.V2ClusterUpdat
 	}
 	return networkType
 }
+
+// Returns true if openshiftVersion is less than targetVersion or undefined
+func openshiftVersionLessThan(targetVersion, openshiftVersion string) bool {
+	isAvailable, err := common.BaseVersionLessThan(targetVersion, openshiftVersion);
```
[the remark this answers is not in the document]
Comment (rccrdpccl) [reply internal/featuresupport/features_networking.go]: yep :sweat: linter complained too, and rightly so

**Review by adriengentil — APPROVED** [review]

**Pull request merged**